pints/tests/test_slice_stepout.py

```diff
@@ -0,0 +1,350 @@
+
+    Please refer to the _slice_stepout.py script in ..\_mcmc\_slice_stepout.py
+    """
+    def test_initialisation(self):
+        """
+        Tests whether all instance attributes are initialised correctly.
+        """
+        # Create mcmc
+        x0 = np.array([2, 4])
+        mcmc = pints.SliceStepoutMCMC(x0)
+
+        # Test attributes initialisation
+        self.assertFalse(mcmc._running)
+        self.assertFalse(mcmc._ready_for_tell)
+        self.assertFalse(mcmc._first_expansion)
+        self.assertFalse(mcmc._interval_found)
+        self.assertFalse(mcmc._set_l)
+        self.assertFalse(mcmc._set_r)
+        self.assertFalse(mcmc._init_left)
+        self.assertFalse(mcmc._init_right)
+
+        self.assertEqual(mcmc._current, None)
+        self.assertEqual(mcmc._current_log_pdf, None)
+        self.assertEqual(mcmc._current_log_y, None)
+        self.assertEqual(mcmc._proposed, None)
+        self.assertEqual(mcmc._l, None)
+        self.assertEqual(mcmc._r, None)
+        self.assertEqual(mcmc._temp_l, None)
+        self.assertEqual(mcmc._temp_r, None)
+        self.assertEqual(mcmc._k, None)
+        self.assertEqual(mcmc._j, None)
+        self.assertEqual(mcmc._fx_l, None)
+        self.assertEqual(mcmc._fx_r, None)
+
+        self.assertEqual(mcmc._m, 50)
+        self.assertEqual(mcmc._active_param_index, 0)
+
+    def test_first_run(self):
+        """
+        Tests the very first run of the sampler.
+        """
+        # Create log pdf
+        log_pdf = pints.toy.GaussianLogPDF([2, 4], [[1, 0], [0, 3]])
+
+        # Create mcmc
+        x0 = np.array([2., 4.])
+        mcmc = pints.SliceStepoutMCMC(x0)
+
+        # Ask should fail if _ready_for_tell flag is True
+        with self.assertRaises(RuntimeError):
+            mcmc._ready_for_tell = True
+            mcmc.ask()
+
+        # Undo changes
+        mcmc._ready_for_tell = False
+
+        # Check whether _running flag becomes True when ask() is called
+        # Check whether first iteration of ask() returns x0
+        self.assertFalse(mcmc._running)
+        self.assertTrue(np.all(mcmc.ask() == x0))
+        self.assertTrue(mcmc._running)
+        self.assertTrue(mcmc._ready_for_tell)
+
+        # Tell should fail when log pdf of x0 is infinite
+        with self.assertRaises(ValueError):
+            fx = np.inf
+            mcmc.tell(fx)
+
+        # Calculate log pdf for x0
+        fx = log_pdf.evaluateS1(x0)[0]
+
+        # Tell should fail when _ready_for_tell is False
+        with self.assertRaises(RuntimeError):
+            mcmc._ready_for_tell = False
+            mcmc.tell(fx)
+
+        # Undo changes
+        mcmc._ready_for_tell = True
+
+        # Test first iteration of tell(). The first point in the chain
+        # should be x0
+        self.assertTrue(np.all(mcmc.tell(fx) == x0))
+
+        # We update the current sample
+        self.assertTrue(np.all(mcmc._current == x0))
```

Comment:
These should all go!

(But you _can_ test this is there are already methods that give you the current sample!)